libsolidity/interface/StandardCompiler.cpp

```diff
@@ -323,10 +323,12 @@ Json::Value formatLinkReferences(std::map<size_t, std::string> const& linkRefere
 	for (auto const& ref: linkReferences)
 	{
 		string const& fullname = ref.second;
+
+		// If the link reference does not contain a colon, assume that the file name is missing and
+		// the whole string represents the library name.
```

Comment:
When is this the case?

Reply:
In Yul. Solidity code will always generate something like `linkersymbol("library.sol:L")` but when you're writing Yul directly you can do just `linkersymbol("L")` because the references are technically just arbitrary free-form strings.

Reply:
Hmm, but then shouldn't we only allow this when `language: "Yul"`? I think it may be a bad idea doing this for the Solidity case, if we have been strict about it so far.

Reply:
Good point. I'll restore the assert for Solidity code in a separate PR (because this one has been merged).